core/src/main/java/com/graphhopper/routing/profiles/MappedDecimalEncodedValue.java

```diff
@@ -43,17 +43,19 @@ public MappedDecimalEncodedValue(String name, Collection<Double> values, double
 
         this.precision = precision;
         // store int-int mapping
-        toValueMap = new int[values.size()];
+        toValueMapArray = new int[values.size()];
         toStorageMap = new IntIntHashMap(values.size());
 
+        List<Double> sortedValues = new ArrayList<>(values);
+        Collections.sort(sortedValues);
         int index = 0;
         IntHashSet dupCheck = new IntHashSet();
-        for (double val : values) {
+        for (double val : sortedValues) {
             int intVal = toInt(val);
             if (!dupCheck.add(intVal)) {
                 throw new IllegalArgumentException("The value " + val + " was converted to " + intVal + " but this already exists. Remove it to improve efficiency.");
```

Comment:
Can we improve this error message? Right now this sounds more like an INFO log statement. We could also mention that one could also increase the precision to fix the error.